components/webac/src/main/java/org/trellisldp/webac/WebAcService.java

```diff
@@ -108,18 +113,44 @@ public WebAcService() {
         this(new NoopResourceService());
     }
 
-    private static List<Authorization> generateDefaultRootAuthorizations() {
-        try (final Graph graph = rdf.createGraph()) {
-            graph.add(rdf.createTriple(rootAuth, ACL.mode, ACL.Read));
-            graph.add(rdf.createTriple(rootAuth, ACL.mode, ACL.Write));
-            graph.add(rdf.createTriple(rootAuth, ACL.mode, ACL.Control));
-            graph.add(rdf.createTriple(rootAuth, ACL.mode, ACL.Append));
-            graph.add(rdf.createTriple(rootAuth, ACL.agentClass, FOAF.Agent));
-            graph.add(rdf.createTriple(rootAuth, ACL.default_, root));
-            graph.add(rdf.createTriple(rootAuth, ACL.accessTo, root));
-            return singletonList(Authorization.from(rootAuth, graph));
+    private static Dataset generateDefaultRootAuthorizations() {
+        final Dataset dataset = rdf.createDataset();
+        dataset.add(rdf.createQuad(Trellis.PreferAccessControl, rootAuth, ACL.mode, ACL.Read));
+        dataset.add(rdf.createQuad(Trellis.PreferAccessControl, rootAuth, ACL.mode, ACL.Write));
+        dataset.add(rdf.createQuad(Trellis.PreferAccessControl, rootAuth, ACL.mode, ACL.Control));
+        dataset.add(rdf.createQuad(Trellis.PreferAccessControl, rootAuth, ACL.mode, ACL.Append));
+        dataset.add(rdf.createQuad(Trellis.PreferAccessControl, rootAuth, ACL.agentClass, FOAF.Agent));
+        dataset.add(rdf.createQuad(Trellis.PreferAccessControl, rootAuth, ACL.default_, root));
+        dataset.add(rdf.createQuad(Trellis.PreferAccessControl, rootAuth, ACL.accessTo, root));
+        return dataset;
+    }
+
+    /**
+     * Initializes the root ACL, if there is no root ACL.
+     */
+    @PostConstruct
+    public void initialize() {
+        try (final Dataset dataset = generateDefaultRootAuthorizations()) {
+            this.resourceService.get(root).thenCompose(res -> initialize(root, res, dataset))
+                .exceptionally(err -> {
+                    LOGGER.warn("Unable to auto-initialize Trellis: {}. See DEBUG log for more info", err.getMessage());
+                    LOGGER.debug("Error auto-initializing Trellis", err);
+                    return null;
+                }).toCompletableFuture().join();
         } catch (final Exception ex) {
-            throw new RuntimeTrellisException("Error closing graph", ex);
+            throw new RuntimeTrellisException("Error closing dataset", ex);
+        }
+    }
+
+    private CompletionStage<Void> initialize(final IRI id, final Resource res, final Dataset dataset) {
+        if (!res.hasAcl()) {
+            LOGGER.info("Initializing root ACL: {}", id);
+            try (final Stream<Quad> quads = res.stream(Trellis.PreferUserManaged)) {
+                quads.forEach(dataset::add);
+            }
+            return this.resourceService.replace(Metadata.builder(res).build(), dataset);
+        } else {
+            return runAsync(() -> { });
```

Comment:
I would return a `CompletableFuture.completedFuture(null)`. Also I would take that out as a constant called `DONE` or the like.

Reply:
Also I would log (at `INFO`) something here, like "No changes made to pre-existing root ACL."

Reply:
I was wondering what the shorthand should be there.. thanks.